### ISSUE TITLE
[JENKINS-47219] Conditional parallel stages are not rendered properly if skipped

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#4476]
<p><b>Problem</b><br/>
Skipped parallel stages show up as "successful" rather than "not built".</p>

<p>They also do not have the expected visual treatment (e.g. fadeout the connected line).</p>

<p><b>Screenshot</b><br/>
 <span class="image-wrap" style=""><a id="39918_thumb" href="https://issues.jenkins.io/secure/attachment/39918/39918_problem.png" title="problem.png" file-preview-type="image" file-preview-id="39918" file-preview-title="problem.png"><img src="https://issues.jenkins.io/secure/thumbnail/39918/_thumb_39918.png" style="border: 0px solid black" role="presentation"/></a></span> </p>

<p><b>Jenkinsfile</b></p>

<pre>
pipeline {
    agent any
    stages {
stage(<span class="code-quote">'Run Tests'</span>) {
    parallel {
stage(<span class="code-quote">'Test On Windows'</span>) {
    when {
      branch <span class="code-quote">'cake'</span>
    }
    steps {
echo <span class="code-quote">'hello world'</span>
    }
}
stage(<span class="code-quote">'Test On Linux'</span>) {
    steps {
echo <span class="code-quote">'hello world'</span>
    }
}
    }
}
    }
}
</pre>

<p><b>Log</b></p>

```
Started by user admin
[Pipeline] node
Running on master in /Users/jdumay/.jenkins/workspace/JENKINS-47219
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Run Tests)
[Pipeline] parallel
[Pipeline] [Test On Windows] { (Branch: Test On Windows)
[Pipeline] [Test On Linux] { (Branch: Test On Linux)
[Pipeline] [Test On Windows] stage
[Pipeline] [Test On Windows] { (Test On Windows)
[Pipeline] [Test On Linux] stage
[Pipeline] [Test On Linux] { (Test On Linux)
Stage 'Test On Windows' skipped due to when conditional
[Pipeline] [Test On Windows] }
[Pipeline] [Test On Windows] // stage
[Pipeline] [Test On Windows] }
[Pipeline] [Test On Linux] echo
[Test On Linux] hello world
[Pipeline] [Test On Linux] }
[Pipeline] [Test On Linux] // stage
[Pipeline] [Test On Linux] }
[Pipeline] // parallel
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
Finished: SUCCESS

```

<p><b>Original Request</b><br/>
If conditional step in parallel is skipped, it is still rendered as executed, just without steps.</p>

<p>See attached images.</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=quas">quas</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219" target="_blank">Conditional parallel stages are not rendered properly if skipped</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers">cliffmeyers</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Critical
<li><b>component(s)</b>: blueocean-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2018-01-08T21:37:56+00:00
<li><b>votes</b>: 6
<li><b>watchers</b>: 13
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p><b>Problem</b>
Skipped parallel stages show up as "successful" rather than "not built".</p>

<p>They also do not have the expected visual treatment (e.g. fadeout the connected line).</p>

<p><b>Screenshot</b>
 <span class="image-wrap" style=""><a id="39918_thumb" href="https://issues.jenkins.io/secure/attachment/39918/39918_problem.png" title="problem.png" file-preview-type="image" file-preview-id="39918" file-preview-title="problem.png"><img src="https://issues.jenkins.io/secure/thumbnail/39918/_thumb_39918.png" style="border: 0px solid black" role="presentation"/></a></span> </p>

<p><b>Jenkinsfile</b></p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">pipeline {
    agent any
    stages {
        stage(<span class="code-quote">'Run Tests'</span>) {
            parallel {
                stage(<span class="code-quote">'Test On Windows'</span>) {
                    when {
                      branch <span class="code-quote">'cake'</span>
                    }
                    steps {
                        echo <span class="code-quote">'hello world'</span>
                    }
                }
                stage(<span class="code-quote">'Test On Linux'</span>) {
                    steps {
                        echo <span class="code-quote">'hello world'</span>
                    }
                }
            }
        }
    }
}
</pre>
</div></div>

<p><b>Log</b></p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>Started by user admin
[Pipeline] node
Running on master in /Users/jdumay/.jenkins/workspace/JENKINS-47219
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Run Tests)
[Pipeline] parallel
[Pipeline] [Test On Windows] { (Branch: Test On Windows)
[Pipeline] [Test On Linux] { (Branch: Test On Linux)
[Pipeline] [Test On Windows] stage
[Pipeline] [Test On Windows] { (Test On Windows)
[Pipeline] [Test On Linux] stage
[Pipeline] [Test On Linux] { (Test On Linux)
Stage 'Test On Windows' skipped due to when conditional
[Pipeline] [Test On Windows] }
[Pipeline] [Test On Windows] // stage
[Pipeline] [Test On Windows] }
[Pipeline] [Test On Linux] echo
[Test On Linux] hello world
[Pipeline] [Test On Linux] }
[Pipeline] [Test On Linux] // stage
[Pipeline] [Test On Linux] }
[Pipeline] // parallel
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
Finished: SUCCESS
</pre>
</div></div>

<p><b>Original Request</b>
If conditional step in parallel is skipped, it is still rendered as executed, just without steps.</p>

<p>See attached images.</p></pre>
</details>
</details>
<details><summary><i>8 attachments</i></summary>

- [2017-12-28-1-skipped-parallel.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/40914/2017-12-28-1-skipped-parallel.png)
  > ![2017-12-28-1-skipped-parallel.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/40914/2017-12-28-1-skipped-parallel.png)
- [Jenkins-pipeline-failure-skipped-parallel-steps-green.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/41030/Jenkins-pipeline-failure-skipped-parallel-steps-green.png)
  > ![Jenkins-pipeline-failure-skipped-parallel-steps-green.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/41030/Jenkins-pipeline-failure-skipped-parallel-steps-green.png)
- [expected.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/39899/expected.png)
  > ![expected.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/39899/expected.png)
- [expected2.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/39901/expected2.png)
  > ![expected2.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/39901/expected2.png)
- [image-2017-12-09-02-06-05-091.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/40628/image-2017-12-09-02-06-05-091.png)
  > ![image-2017-12-09-02-06-05-091.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/40628/image-2017-12-09-02-06-05-091.png)
- [problem.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/39918/problem.png)
  > ![problem.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/39918/problem.png)
- [problem.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/39919/problem.png)
  > ![problem.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/39919/problem.png)
- [result.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/39900/result.png)
  > ![result.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/39900/result.png)
</details>

**Comment by jenkins-infra-bot** [comment]
- _Remote links associated with this issue:_


  - [PR#​16](https://github.com/jenkinsci/pipeline-graph-analysis-plugin/pull/16)
  - [PR#​1610](https://github.com/jenkinsci/blueocean-plugin/pull/1610)
  - [Page](https://wiki.jenkins.io/pages/viewpage.action?pageId=98403475)

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>quas:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=315511&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-315511">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>as from the look of it you can already tell its parallel, I don't think it is needed to caption its name, that allows to moving stage labels above their state, expected2 image demonstrates it</p>
</pre>
</details>

</ul>
</details>

<p>as from the look of it you can already tell its parallel, I don't think it is needed to caption its name, that allows to moving stage labels above their state, expected2 image demonstrates it</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=315686&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-315686">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=quas" class="user-hover" rel="quas">quas</a> could you please post an example Jenkinsfile that reproduces this issue? </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=quas" class="user-hover" rel="quas">quas</a> could you please post an example Jenkinsfile that reproduces this issue? </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>quas:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=315764&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-315764">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Here is frame I'm using, you will need to approve two scripts.</p>

<p> </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"><span class="code-object">String</span> ProcessChoices = [<span class="code-quote">"None"</span>, <span class="code-quote">"Debug"</span>, <span class="code-quote">"Release"</span>, <span class="code-quote">"Debug and Release"</span>, <span class="code-quote">"Release and Publish"</span>, <span class="code-quote">"All"</span>].join(<span class="code-quote">"\n"</span>)

<span class="code-keyword">enum</span> Configs { Debug, Release }

def Build(<span class="code-object">String</span> Workspace, Configs Config) { echo <span class="code-quote">"BUILD ${Workspace} ${Config}"</span> }

def Publish(<span class="code-object">String</span> Workspace) { echo <span class="code-quote">"PUBLISH ${Workspace}"</span> }

pipeline {
  agent any

  environment {
    DebugPattern = <span class="code-quote">"Debug|Debug and Release|All"</span>
    ReleasePattern = <span class="code-quote">"Release|Debug and Release|Release and Publish|Release and Test|All"</span>
    PublishPattern = <span class="code-quote">"Release and Publish|All"</span>
  }

  parameters {
    choice(name: <span class="code-quote">'<span class="code-object">Process</span>'</span>, choices: ProcessChoices)
  }

  stages {
    stage (<span class="code-quote">'parallel'</span>) {
      parallel {
        stage (<span class="code-quote">"Debug"</span>) {
          when { expression { params.<span class="code-object">Process</span>.matches(env.DebugPattern)}}
          steps { Build(WORKSPACE, Configs.Debug) }
        }

        stage (<span class="code-quote">"Release"</span>) {
          when { expression { params.<span class="code-object">Process</span>.matches(env.ReleasePattern)}}
          steps { Build(WORKSPACE, Configs.Release) }
        }
      }
    }

    stage (<span class="code-quote">"Publish"</span>) {
      when { expression { params.<span class="code-object">Process</span>.matches(env.PublishPattern)}}
      steps { Publish(WORKSPACE) }
    }
  }
}
</pre>
</div></div>
<p> </p>

<p> </p>
</pre>
</details>

</ul>
</details>

<p>Here is frame I'm using, you will need to approve two scripts.</p>

<p> </p>

<pre>
<span class="code-object">String</span> ProcessChoices = [<span class="code-quote">"None"</span>, <span class="code-quote">"Debug"</span>, <span class="code-quote">"Release"</span>, <span class="code-quote">"Debug and Release"</span>, <span class="code-quote">"Release and Publish"</span>, <span class="code-quote">"All"</span>].join(<span class="code-quote">"\n"</span>)

<span class="code-keyword">enum</span> Configs { Debug, Release }

def Build(<span class="code-object">String</span> Workspace, Configs Config) { echo <span class="code-quote">"BUILD ${Workspace} ${Config}"</span> }

def Publish(<span class="code-object">String</span> Workspace) { echo <span class="code-quote">"PUBLISH ${Workspace}"</span> }

pipeline {
  agent any

  environment {
    DebugPattern = <span class="code-quote">"Debug|Debug and Release|All"</span>
    ReleasePattern = <span class="code-quote">"Release|Debug and Release|Release and Publish|Release and Test|All"</span>
    PublishPattern = <span class="code-quote">"Release and Publish|All"</span>
  }

  parameters {
    choice(name: <span class="code-quote">'<span class="code-object">Process</span>'</span>, choices: ProcessChoices)
  }

  stages {
    stage (<span class="code-quote">'parallel'</span>) {
      parallel {
        stage (<span class="code-quote">"Debug"</span>) {
          when { expression { params.<span class="code-object">Process</span>.matches(env.DebugPattern)}}
          steps { Build(WORKSPACE, Configs.Debug) }
        }

        stage (<span class="code-quote">"Release"</span>) {
          when { expression { params.<span class="code-object">Process</span>.matches(env.ReleasePattern)}}
  steps { Build(WORKSPACE, Configs.Release) }
        }
      }
    }

    stage (<span class="code-quote">"Publish"</span>) {
      when { expression { params.<span class="code-object">Process</span>.matches(env.PublishPattern)}}
      steps { Publish(WORKSPACE) }
    }
  }
}
</pre>
<p> </p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>quas:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=315768&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-315768">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Maybe it is forth splitting it into two, as:</p>
<ul>
	<li>a bug - that visualises not executed parallel stage as successful (could be fixed with this purple status we had previously)</li>
	<li>a feature - to make it look so fancy as non parallel skipped stages</li>
</ul>


<p>I thing the bug should be fixed asap as it is misleading, feature is just nice to have.</p>
</pre>
</details>

</ul>
</details>

<p>Maybe it is forth splitting it into two, as:</p>
<ul>
	<li>a bug - that visualises not executed parallel stage as successful (could be fixed with this purple status we had previously)</li>
	<li>a feature - to make it look so fancy as non parallel skipped stages</li>
</ul>


<p>I thing the bug should be fixed asap as it is misleading, feature is just nice to have.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=315834&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-315834">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=quas" class="user-hover" rel="quas">quas</a> I suspect its roughly about the same amount of effort from our point of view. I hope to have a fix out soon.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=quas" class="user-hover" rel="quas">quas</a> I suspect its roughly about the same amount of effort from our point of view. I hope to have a fix out soon.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/nroth.png" title="nroth's avatar" /> nroth:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=319128&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-319128">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This is a major issue for me too. It just looks misleading to have jobs show up as successful if they were skipped due to an earlier failure or when condition.</p>
</pre>
</details>

</ul>
</details>

<p>This is a major issue for me too. It just looks misleading to have jobs show up as successful if they were skipped due to an earlier failure or when condition.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>yogesh_ar:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=322296&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-322296">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This is also a major issues for us. It's very misleading when parallel stages are skipped but marked as green. Sample is attached for reference. Is there any plan to fix the issue?</p>

<p> </p>

<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/40628/40628_image-2017-12-09-02-06-05-091.png" style="border: 0px solid black" /></span></p>
</pre>
</details>

</ul>
</details>

<p>This is also a major issues for us. It's very misleading when parallel stages are skipped but marked as green. Sample is attached for reference. Is there any plan to fix the issue?</p>

<p> </p>

<p><span class="image-wrap" style=""><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/40628/image-2017-12-09-02-06-05-091.png" style="border: 0px solid black" /></span></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>quas:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=322893&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-322893">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Also I think it misleads avg times, as under some condition step takes an hour but when skipped it takes 0, so on avg it takes 30 minutes <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>Also I think it misleads avg times, as under some condition step takes an hour but when skipped it takes 0, so on avg it takes 30 minutes <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=323153&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-323153">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> mind taking a look?</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> mind taking a look?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=323325&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-323325">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The root cause is that the status computed for the underlying FlowNode is incorrect. In pipeline-graph-analysis-plugin, the StatusAndTiming.computeChunkStatus2 method will only return "NOT_EXECUTED" when the FlowNode has "NotExecutedNodeAction" attached . In this case however, it instead has a "TagsAction" with key/value of "STAGE_STATUS" -&gt; "SKIPPED_FOR_CONDITIONAL." Chatting w/ Sam, I think it's some subtlety with how the declarative and/or parallel stages are handled.</p>

<p>I am working on a PR for that upstream plugin which adds the proper check.</p>
</pre>
</details>

</ul>
</details>

<p>The root cause is that the status computed for the underlying FlowNode is incorrect. In pipeline-graph-analysis-plugin, the StatusAndTiming.computeChunkStatus2 method will only return "NOT_EXECUTED" when the FlowNode has "NotExecutedNodeAction" attached . In this case however, it instead has a "TagsAction" with key/value of "STAGE_STATUS" -> "SKIPPED_FOR_CONDITIONAL." Chatting w/ Sam, I think it's some subtlety with how the declarative and/or parallel stages are handled.</p>

<p>I am working on a PR for that upstream plugin which adds the proper check.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=323329&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-323329">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay" class="user-hover" rel="jamesdumay">jamesdumay</a> additional requirements question: suppose that both of the parallel stages in the enclosing stage were skipped: is the top-level stage's status equal to "SUCCESS" or "NOT_BUILT" ?</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay" class="user-hover" rel="jamesdumay">jamesdumay</a> additional requirements question: suppose that both of the parallel stages in the enclosing stage were skipped: is the top-level stage's status equal to "SUCCESS" or "NOT_BUILT" ?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=323834&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-323834">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The fix in PR#16 in "pipeline-graph-analysis-plugin" has resolved the issue with the data being reported incorrectly. For the sample Jenkinsfile in the description, here is the REST response (abbreviated slightly for readability)</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">[
    {
        <span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.<span class="code-keyword">rest</span>.impl.pipeline.PipelineNodeImpl"</span>,
        <span class="code-quote">"_links"</span>: {},
        <span class="code-quote">"actions"</span>: [],
        <span class="code-quote">"displayDescription"</span>: <span class="code-keyword">null</span>,
        <span class="code-quote">"displayName"</span>: <span class="code-quote">"Run Tests"</span>,
        <span class="code-quote">"durationInMillis"</span>: 32,
        <span class="code-quote">"id"</span>: <span class="code-quote">"6"</span>,
        <span class="code-quote">"input"</span>: <span class="code-keyword">null</span>,
        <span class="code-quote">"result"</span>: <span class="code-quote">"SUCCESS"</span>,
        <span class="code-quote">"startTime"</span>: <span class="code-quote">"2017-12-20T13:10:40.484-0500"</span>,
        <span class="code-quote">"state"</span>: <span class="code-quote">"FINISHED"</span>,
        <span class="code-quote">"type"</span>: <span class="code-quote">"STAGE"</span>,
        <span class="code-quote">"causeOfBlockage"</span>: <span class="code-keyword">null</span>,
        <span class="code-quote">"edges"</span>: []
    },
    {
        <span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.<span class="code-keyword">rest</span>.impl.pipeline.PipelineNodeImpl"</span>,
        <span class="code-quote">"_links"</span>: {},
        <span class="code-quote">"actions"</span>: [],
        <span class="code-quote">"displayDescription"</span>: <span class="code-keyword">null</span>,
        <span class="code-quote">"displayName"</span>: <span class="code-quote">"Test On Linux"</span>,
        <span class="code-quote">"durationInMillis"</span>: 30,
        <span class="code-quote">"id"</span>: <span class="code-quote">"10"</span>,
        <span class="code-quote">"input"</span>: <span class="code-keyword">null</span>,
        <span class="code-quote">"result"</span>: <span class="code-quote">"SUCCESS"</span>,
        <span class="code-quote">"startTime"</span>: <span class="code-quote">"2017-12-20T13:10:40.486-0500"</span>,
        <span class="code-quote">"state"</span>: <span class="code-quote">"FINISHED"</span>,
        <span class="code-quote">"type"</span>: <span class="code-quote">"PARALLEL"</span>,
        <span class="code-quote">"causeOfBlockage"</span>: <span class="code-keyword">null</span>,
        <span class="code-quote">"edges"</span>: []
    },
    {
        <span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.<span class="code-keyword">rest</span>.impl.pipeline.PipelineNodeImpl"</span>,
        <span class="code-quote">"_links"</span>: {},
        <span class="code-quote">"actions"</span>: [],
        <span class="code-quote">"displayDescription"</span>: <span class="code-keyword">null</span>,
        <span class="code-quote">"displayName"</span>: <span class="code-quote">"Test On Windows"</span>,
        <span class="code-quote">"durationInMillis"</span>: 31,
        <span class="code-quote">"id"</span>: <span class="code-quote">"9"</span>,
        <span class="code-quote">"input"</span>: <span class="code-keyword">null</span>,
        <span class="code-quote">"result"</span>: <span class="code-quote">"NOT_BUILT"</span>,
        <span class="code-quote">"startTime"</span>: <span class="code-quote">"2017-12-20T13:10:40.485-0500"</span>,
        <span class="code-quote">"state"</span>: <span class="code-quote">"NOT_BUILT"</span>,
        <span class="code-quote">"type"</span>: <span class="code-quote">"PARALLEL"</span>,
        <span class="code-quote">"causeOfBlockage"</span>: <span class="code-keyword">null</span>,
        <span class="code-quote">"edges"</span>: []
    }
]
</pre>
</div></div>

<p>This is rendered in the UI as follows:</p>

<p> <span class="image-wrap" style=""><a id="40914_thumb" href="https://issues.jenkins.io/secure/attachment/40914/40914_2017-12-28-1-skipped-parallel.png" title="2017-12-28-1-skipped-parallel.png" file-preview-type="image" file-preview-id="40914" file-preview-title="2017-12-28-1-skipped-parallel.png"><img src="https://issues.jenkins.io/secure/thumbnail/40914/_thumb_40914.png" style="border: 0px solid black" role="presentation"/></a></span> </p>
</pre>
</details>

</ul>
</details>

<p>The fix in PR#16 in "pipeline-graph-analysis-plugin" has resolved the issue with the data being reported incorrectly. For the sample Jenkinsfile in the description, here is the REST response (abbreviated slightly for readability)</p>


<pre>
[
    {
<span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.<span class="code-keyword">rest</span>.impl.pipeline.PipelineNodeImpl"</span>,
<span class="code-quote">"_links"</span>: {},
<span class="code-quote">"actions"</span>: [],
<span class="code-quote">"displayDescription"</span>: <span class="code-keyword">null</span>,
<span class="code-quote">"displayName"</span>: <span class="code-quote">"Run Tests"</span>,
<span class="code-quote">"durationInMillis"</span>: 32,
<span class="code-quote">"id"</span>: <span class="code-quote">"6"</span>,
<span class="code-quote">"input"</span>: <span class="code-keyword">null</span>,
<span class="code-quote">"result"</span>: <span class="code-quote">"SUCCESS"</span>,
<span class="code-quote">"startTime"</span>: <span class="code-quote">"2017-12-20T13:10:40.484-0500"</span>,
<span class="code-quote">"state"</span>: <span class="code-quote">"FINISHED"</span>,
<span class="code-quote">"type"</span>: <span class="code-quote">"STAGE"</span>,
<span class="code-quote">"causeOfBlockage"</span>: <span class="code-keyword">null</span>,
<span class="code-quote">"edges"</span>: []
    },
    {
<span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.<span class="code-keyword">rest</span>.impl.pipeline.PipelineNodeImpl"</span>,
<span class="code-quote">"_links"</span>: {},
<span class="code-quote">"actions"</span>: [],
<span class="code-quote">"displayDescription"</span>: <span class="code-keyword">null</span>,
<span class="code-quote">"displayName"</span>: <span class="code-quote">"Test On Linux"</span>,
<span class="code-quote">"durationInMillis"</span>: 30,
<span class="code-quote">"id"</span>: <span class="code-quote">"10"</span>,
<span class="code-quote">"input"</span>: <span class="code-keyword">null</span>,
<span class="code-quote">"result"</span>: <span class="code-quote">"SUCCESS"</span>,
<span class="code-quote">"startTime"</span>: <span class="code-quote">"2017-12-20T13:10:40.486-0500"</span>,
<span class="code-quote">"state"</span>: <span class="code-quote">"FINISHED"</span>,
<span class="code-quote">"type"</span>: <span class="code-quote">"PARALLEL"</span>,
<span class="code-quote">"causeOfBlockage"</span>: <span class="code-keyword">null</span>,
<span class="code-quote">"edges"</span>: []
    },
    {
<span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.<span class="code-keyword">rest</span>.impl.pipeline.PipelineNodeImpl"</span>,
<span class="code-quote">"_links"</span>: {},
<span class="code-quote">"actions"</span>: [],
<span class="code-quote">"displayDescription"</span>: <span class="code-keyword">null</span>,
<span class="code-quote">"displayName"</span>: <span class="code-quote">"Test On Windows"</span>,
<span class="code-quote">"durationInMillis"</span>: 31,
<span class="code-quote">"id"</span>: <span class="code-quote">"9"</span>,
<span class="code-quote">"input"</span>: <span class="code-keyword">null</span>,
<span class="code-quote">"result"</span>: <span class="code-quote">"NOT_BUILT"</span>,
<span class="code-quote">"startTime"</span>: <span class="code-quote">"2017-12-20T13:10:40.485-0500"</span>,
<span class="code-quote">"state"</span>: <span class="code-quote">"NOT_BUILT"</span>,
<span class="code-quote">"type"</span>: <span class="code-quote">"PARALLEL"</span>,
<span class="code-quote">"causeOfBlockage"</span>: <span class="code-keyword">null</span>,
<span class="code-quote">"edges"</span>: []
    }
]
</pre>

<p>This is rendered in the UI as follows:</p>

<p> <span class="image-wrap" style=""><a id="40914_thumb" href="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/40914/2017-12-28-1-skipped-parallel.png" title="2017-12-28-1-skipped-parallel.png" file-preview-type="image" file-preview-id="40914" file-preview-title="2017-12-28-1-skipped-parallel.png"><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/40914/2017-12-28-1-skipped-parallel.png" style="border: 0px solid black" role="presentation"/></a></span> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=323835&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-323835">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay" class="user-hover" rel="jamesdumay">jamesdumay</a> or <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale" class="user-hover" rel="michaelneale">michaelneale</a> please let me know if the screenie attached above is what we expected for a single skipped parallel.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay" class="user-hover" rel="jamesdumay">jamesdumay</a> or <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale" class="user-hover" rel="michaelneale">michaelneale</a> please let me know if the screenie attached above is what we expected for a single skipped parallel.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=324012&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-324012">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Seems fine to me! (James hasn’t replied though) </p>
</pre>
</details>

</ul>
</details>

<p>Seems fine to me! (James hasn’t replied though) </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=324067&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-324067">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Sam Van Oort
Path:
 pom.xml
 src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StageStatus.java
 src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTiming.java
 src/test/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StageTest.java
 src/test/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTimingTest.java
<a href="http://jenkins-ci.org/commit/pipeline-graph-analysis-plugin/f1b74e91bbb9824618c4848f319b438c302eb396" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/pipeline-graph-analysis-plugin/f1b74e91bbb9824618c4848f319b438c302eb396</a>
Log:
  Merge pull request #16 from cliffmeyers/bug/<a href="https://issues.jenkins.io/browse/JENKINS-47219" title="Conditional parallel stages are not rendered properly if skipped" class="issue-link" data-issue-key="JENKINS-47219"><del>JENKINS-47219</del></a>-skipped-parallel-stage-wrong-status</p>

<p>Bug/jenkins 47219 skipped parallel stage wrong status</p>


<p>Compare: <a href="https://github.com/jenkinsci/pipeline-graph-analysis-plugin/compare/7ea371dc90fe...f1b74e91bbb9" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/pipeline-graph-analysis-plugin/compare/7ea371dc90fe...f1b74e91bbb9</a></p>

</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Sam Van Oort<br/>
Path:<br/>
 pom.xml<br/>
 src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StageStatus.java<br/>
 src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTiming.java<br/>
 src/test/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StageTest.java<br/>
 src/test/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTimingTest.java<br/>
<a href="http://jenkins-ci.org/commit/pipeline-graph-analysis-plugin/f1b74e91bbb9824618c4848f319b438c302eb396" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/pipeline-graph-analysis-plugin/f1b74e91bbb9824618c4848f319b438c302eb396</a><br/>
Log:<br/>
  Merge pull request #16 from cliffmeyers/bug/<a href="https://issues.jenkins.io/browse/JENKINS-47219" title="Conditional parallel stages are not rendered properly if skipped" class="issue-link" data-issue-key="JENKINS-47219"><del>JENKINS-47219</del></a>-skipped-parallel-stage-wrong-status</p>

<p>Bug/jenkins 47219 skipped parallel stage wrong status</p>


<p>Compare: <a href="https://github.com/jenkinsci/pipeline-graph-analysis-plugin/compare/7ea371dc90fe...f1b74e91bbb9" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/pipeline-graph-analysis-plugin/compare/7ea371dc90fe...f1b74e91bbb9</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=324144&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-324144">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> LGTM</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> LGTM</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>vholer:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=324255&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-324255">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Is it possible to properly visualize the skipped stages also in the scripted pipeline somehow? Thank you.</p>
</pre>
</details>

</ul>
</details>

<p>Is it possible to properly visualize the skipped stages also in the scripted pipeline somehow? Thank you.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/thdepauw.jpg" title="thdepauw's avatar" /> thdepauw:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=324413&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-324413">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We tested the latest version (1.6) of the pipeline-graph-analysis-plugin on our pipeline (which has similar structure as example in the description) and noticed the indication for skipped steps is incorrect when the step is "skipped due to earlier failure(s)".</p>

<p><span class="image-wrap" style=""><a id="41030_thumb" href="https://issues.jenkins.io/secure/attachment/41030/41030_Jenkins-pipeline-failure-skipped-parallel-steps-green.png" title="Jenkins-pipeline-failure-skipped-parallel-steps-green.png" file-preview-type="image" file-preview-id="41030" file-preview-title="Jenkins-pipeline-failure-skipped-parallel-steps-green.png"><img src="https://issues.jenkins.io/secure/thumbnail/41030/_thumb_41030.png" style="border: 0px solid black" role="presentation"/></a></span></p>

<p>When the previous stages succeed, the visual indication for the skipped steps is correct.</p>
</pre>
</details>

</ul>
</details>

<p>We tested the latest version (1.6) of the pipeline-graph-analysis-plugin on our pipeline (which has similar structure as example in the description) and noticed the indication for skipped steps is incorrect when the step is "skipped due to earlier failure(s)".</p>

<p><span class="image-wrap" style=""><a id="41030_thumb" href="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/41030/Jenkins-pipeline-failure-skipped-parallel-steps-green.png" title="Jenkins-pipeline-failure-skipped-parallel-steps-green.png" file-preview-type="image" file-preview-id="41030" file-preview-title="Jenkins-pipeline-failure-skipped-parallel-steps-green.png"><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/41030/Jenkins-pipeline-failure-skipped-parallel-steps-green.png" style="border: 0px solid black" role="presentation"/></a></span></p>

<p>When the previous stages succeed, the visual indication for the skipped steps is correct.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=324537&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-324537">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=thdepauw" class="user-hover" rel="thdepauw">thdepauw</a> can you please provide a simplified Jenkinsfile to reproduce your problem and file a new bug report?</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=thdepauw" class="user-hover" rel="thdepauw">thdepauw</a> can you please provide a simplified Jenkinsfile to reproduce your problem and file a new bug report?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=324612&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-324612">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Version of pipeline-graph-analysis-plugin will be bumped to 1.6 on master, so the next release of Blue Ocean should include this fix by default.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=thdepauw" class="user-hover" rel="thdepauw">thdepauw</a> please cc me on the other issue once it's filed, thank you.</p>
</pre>
</details>

</ul>
</details>

<p>Version of pipeline-graph-analysis-plugin will be bumped to 1.6 on master, so the next release of Blue Ocean should include this fix by default.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=thdepauw" class="user-hover" rel="thdepauw">thdepauw</a> please cc me on the other issue once it's filed, thank you.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/thdepauw.jpg" title="thdepauw's avatar" /> thdepauw:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47219?focusedId=324815&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-324815">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Created <a href="https://issues.jenkins.io/browse/JENKINS-48884" title="Conditional parallel stages are not rendered properly if skipped by failed previous stage" class="issue-link" data-issue-key="JENKINS-48884"><del>JENKINS-48884</del></a> for my issue reported above.</p>
</pre>
</details>

</ul>
</details>

<p>Created <a href="https://issues.jenkins.io/browse/JENKINS-48884" title="Conditional parallel stages are not rendered properly if skipped by failed previous stage" class="issue-link" data-issue-key="JENKINS-48884"><del>JENKINS-48884</del></a> for my issue reported above.</p>
